src/server/store.rs

```diff
@@ -12,13 +12,13 @@ use super::types::LspError;
 // Url is parsed and validated prior to this function. `unwrap` here
 // is okay.
 #[allow(clippy::unwrap_used)]
-fn url_to_key_val(url: &lsp::Url) -> (String, String) {
+fn url_to_key_val(url: &lsp::Url) -> (String, String, String) {
     let path = Path::new(url.path());
 
     let parent: String = path.parent().unwrap().display().to_string();
     let filename = path.file_name().unwrap().to_str().unwrap();
 
-    (parent, filename.into())
+    (url.scheme().to_owned(), parent, filename.into())
```

Comment:
choose to update here, because longterm we may decide to update the path matching elsewhere.